### PR TITLE
Added Shaker Sort in C

### PR DESCRIPTION
**Pull request by rohanbabbar04**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
Implemented Shaker sort In C
### Checklist

- [✅] I've read the contribution guidelines.
- [✅] I've checked the issue list before deciding what to submit.
- [✅] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Added #2235

### CODE REVIEW

File: C/sort/shaker_sort.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+   Enter the number of elements: 13
+   Enter the elements to be sorted:86 74 -2 -9 67 25 33 86 17 -8 42 10 0
+   Sorted Array:-9 -8 -2 0 10 17 25 33 42 67 74 86 86  
+*/
+/* *****************Complexities***************
+   Worst case performance:O(n^2)
+   Best case performance:O(n)
+   Average performance: O(n^2)
+   Worst Case Space complexity: O(1) 
+*/
```
Comment (HarshCasper) [inline C/sort/shaker_sort.c]: Add a new line at the EOF

File: C/sort/shaker_sort.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      }
+}
+void swap(int *x,int *y)
+{
+     int temp;
+     temp=*x;
+     *x=*y;
+     *y=temp;
+}
+/* *****************Sample Output*************
```
Comment (HarshCasper) [inline C/sort/shaker_sort.c]: Remove the asterisks.

File: C/sort/shaker_sort.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+   Enter the number of elements: 13
+   Enter the elements to be sorted:86 74 -2 -9 67 25 33 86 17 -8 42 10 0
+   Sorted Array:-9 -8 -2 0 10 17 25 33 42 67 74 86 86  
+*/
+/* *****************Complexities***************
```
Comment (HarshCasper) [inline C/sort/shaker_sort.c]: Remove the asterisks.

File: C/sort/shaker_sort.c [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+         num2=num2+1;
+      }
+}
+void swap(int *x,int *y)
+{
+     int temp;
+     temp=*x;
+     *x=*y;
+     *y=temp;
+}
```
Comment (HarshCasper) [inline C/sort/shaker_sort.c]: Format the Code using a C Code Formatter

File: C/sort/shaker_sort.c [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                {           if(arr[i]>arr[i+1])
+                                              swap(&arr[i],&arr[i+1]);
+                                              }
+         num1=num1-1;
+         for(i=num1;i>=num2;i--)
+                 {          if(arr[i]<arr[i-1])
+                                              swap(&arr[i],&arr[i-1]);
+                                              }
+         num2=num2+1;
+      }
```
Comment (HarshCasper) [inline C/sort/shaker_sort.c]: Why is the indentation so mis-matched? Please use a proper indentation pattern across the repo

File: C/sort/shaker_sort.c [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    printf("Sorted Array:");
+    for(i=0;i<num;i++)
+             {         printf("%d ",arr[i]);
+                      }
+    getch();
+    return 0;
+}
+void shakersort(int arr[],int num)
+{
+     int num1,num2,i;
```
Comment (HarshCasper) [inline C/sort/shaker_sort.c]: Bad variable names. Use better ones

File: C/sort/shaker_sort.c [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+              {        printf("%d ",arr[i]);
+                      }
+    shakersort(&arr[0],num);
+    printf("\n");
+    printf("Sorted Array:");
+    for(i=0;i<num;i++)
+             {         printf("%d ",arr[i]);
+                      }
+    getch();
+    return 0;
```
Comment (HarshCasper) [inline C/sort/shaker_sort.c]: Improper indentation and formatting. Fix it

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C/sort/shaker_sort.c [hunk 8]
```diff
@@ -0,0 +1,77 @@
+#include <stdio.h>
```
Comment (HarshCasper) [inline C/sort/shaker_sort.c:4]: Add a beginner pitch to the Code

File: C/sort/shaker_sort.c [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+	printf("Sorted Array:");
+	for (i = 0; i < num; i++)
+	{
+		printf("%d ", arr[i]);
+	}
+
+	return 0;
+}
+
+void shakersort(int arr[], int num)
```
Comment (HarshCasper) [inline C/sort/shaker_sort.c:35]: Add proper function descriptions

File: C/sort/shaker_sort.c [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+		{
+			if (arr[i] < arr[i - 1])
+				swap(&arr[i], &arr[i - 1]);
+		}
+
+		start = start + 1;
+	}
+}
+
+void swap(int *x, int *y)
```
Comment (HarshCasper) [inline C/sort/shaker_sort.c:60]: Add proper function descriptions

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by rohanbabbar04** [comment]
I have modified the changes do review the PR @HarshCasper

**Comment by rohanbabbar04** [comment]
Modified the changes do review @HarshCasper

File: C/README.md [hunk 11]
```diff
@@ -12,6 +12,7 @@
 ## CP
 
 - [Activity Selection](cp/Activity_selection.c)
+- [Adjacent Element Product](cp/Adjacent_Element_Product.c)
```
Comment (HarshCasper) [inline C/README.md]: It doesn't exist in your commit. Revert this

**Review by HarshCasper — CHANGES_REQUESTED** [review]
One tiny change. Otherwise looks good to merge

**Comment by rohanbabbar04** [comment]
I have made the changes you can now merge @HarshCasper

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**